### DISCUSSION TITLE
Wake On Lan using an ESP32 Telegram bot

### DISCUSSION THREAD

**Discussion by daniele-salvagni** [daniele-salvagni/dan.salvagni.io#6] [Comments]
**Comments for the blog post:** https://dsalvagni.com/b/embedded-telegram-bot-for-wake-on-lan-pc/

**Comment by munish4u911ex** [comment]
I recently had my hands on an ESP8266 board and had exactly this requirement with my laptop, so I googled it and arrived here. It seems that it is almost what I need, but I have a question: does it require a _wired_ LAN connection from the router / ESP32 to the PC's ethernet port on which the WOL packed be transmitted?
I ask this because in my case, the router is physically away from the PC and a wired Ethernet connection is not feasible, so can it be done that the ESP32 connects somehow to the PC's Ethernet port and directly relays the WOL packet?
Thanks in advance for your help!

**Reply by daniele-salvagni** [reply]
Hi, some laptops have support for WiFi WOL but it's quite rare and it usually just works from a sleep state (not if powered off) so in general you need a wired connection.

The easiest option in your case would be to use something like a SwitchBot to physically press the power button of your laptop. If your laptop has a dedicated Ethernet port you could also try to connect it directly to the ESP32 by using an Ethernet module with a crossover cable but it won't be as straightforward.

**Reply by hobala** [reply]
The question is whether the ESP board has to be connected via LAN being able to send a magic packet to the LAN network card of the computer that should be startet. Would this also work from ESP32 Wifi to PC LAN?

**Reply by daniele-salvagni** [reply]
@hobala yes, the ESP will be able to send the magic packet via WiFi, while the receiving PC should be connected via Ethernet cable to the same local network.

**Comment by nadja6** [comment]
Hello, great project, I even bought M5atom and used everything you created.
I wanted to start two computers with an independent wol1/wol2 command, preferably with a button, unfortunately I can't program it, could you help me modify the code?
Regards
